File: VB/GridControlWithFindPanel/Custom GridView/MyGridViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System.Drawing
Imports DevExpress.XtraGrid.Views.Grid

Namespace GridControlWithFindPanel
	Public Class MyGridViewInfo
		Inherits GridViewInfo
		Public Sub New(ByVal gridView As GridView)
			MyBase.New(gridView)
		End Sub

		Public Overrides Sub CalcRects(ByVal bounds As Rectangle, ByVal partital As Boolean)
			Dim inTop As Boolean = (TryCast(Me.View, MyGridView)).ShowFindPanelOnTop
			Dim r As Rectangle = Rectangle.Empty
			Dim myBounds As Rectangle = bounds
			If (Not inTop) Then
				myBounds.Height = bounds.Height - 45
			End If
			ViewRects.Bounds = myBounds
			ViewRects.Scroll = CalcScrollRect()
			ViewRects.Client = CalcClientRect()
			FilterPanel.Bounds = Rectangle.Empty
			If (Not partital) Then
				CalcRectsConstants()
			End If
			If View.OptionsView.ShowIndicator Then
				ViewRects.IndicatorWidth = Math.Max(View.IndicatorWidth, ViewRects.MinIndicatorWidth)
			End If
			Dim minTop As Integer = ViewRects.Client.Top
			Dim maxBottom As Integer = ViewRects.Client.Bottom
			If View.OptionsView.ShowViewCaption Then
				r = ViewRects.Client
				r.Y = minTop
				r.Height = CalcViewCaptionHeight(ViewRects.Client)
				ViewRects.ViewCaption = r
				minTop = ViewRects.ViewCaption.Bottom
			End If
			If inTop Then
				minTop = UpdateFindControlVisibility(New Rectangle(ViewRects.Client.X, minTop, ViewRects.Client.Width, maxBottom - minTop)).Y
			Else
				UpdateFindControlVisibility(New Rectangle(ViewRects.Client.X, ViewRects.Client.Bottom, ViewRects.Client.Width, maxBottom - minTop))
			End If
			If View.OptionsView.ShowGroupPanel Then
				r = ViewRects.Client
				r.Y = minTop
				r.Height = CalcGroupPanelHeight()
				ViewRects.GroupPanel = r
				minTop = ViewRects.GroupPanel.Bottom
			End If
			minTop = CalcRectsColumnPanel(minTop)
			If View.IsShowFilterPanel Then
				r = ViewRects.Client
				Dim fPanel As Integer = GetFilterPanelHeight()
				r.Y = maxBottom - fPanel
				r.Height = fPanel
				FilterPanel.Bounds = r
				maxBottom = r.Top
			End If
			If View.OptionsView.ShowFooter Then
				r = ViewRects.Client
				r.Height = GetFooterPanelHeight()
				r.Y = maxBottom - r.Height
				ViewRects.Footer = r
				maxBottom = r.Top
			End If
			r = ViewRects.Client
			r.Y = minTop
			r.Height = maxBottom - minTop
			ViewRects.Rows = r
		End Sub
	End Class
End Namespace

File: CS/GridControlWithFindPanel/Custom GridView/MyGridViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System.Drawing;
using DevExpress.XtraGrid.Views.Grid;

namespace GridControlWithFindPanel
{
    public class MyGridViewInfo : GridViewInfo
    {
        public MyGridViewInfo(GridView gridView)
            : base(gridView)
        {
        }

        public override void CalcRects(Rectangle bounds, bool partital)
        {          
            bool inTop = (this.View as MyGridView).ShowFindPanelOnTop;
            Rectangle r = Rectangle.Empty;
            Rectangle myBounds = bounds;
            if (!inTop)
            {
                myBounds.Height = bounds.Height - (this.View as MyGridView).FindPanelHeight;
            }
            ViewRects.Bounds = myBounds;
            ViewRects.Scroll = CalcScrollRect();
            ViewRects.Client = CalcClientRect();
            FilterPanel.Bounds = Rectangle.Empty;
            if (!partital)
            {
                CalcRectsConstants();
            }
            if (View.OptionsView.ShowIndicator)
            {
                ViewRects.IndicatorWidth = Math.Max(View.IndicatorWidth, ViewRects.MinIndicatorWidth);
            }
            int minTop = ViewRects.Client.Top;
            int maxBottom = ViewRects.Client.Bottom;
            if (View.OptionsView.ShowViewCaption)
            {
                r = ViewRects.Client;
                r.Y = minTop;
                r.Height = CalcViewCaptionHeight(ViewRects.Client);
                ViewRects.ViewCaption = r;
                minTop = ViewRects.ViewCaption.Bottom;
            }
            if (inTop)
                minTop = UpdateFindControlVisibility(new Rectangle(ViewRects.Client.X, minTop, ViewRects.Client.Width, maxBottom - minTop), false).Y;
            else
                UpdateFindControlVisibility(new Rectangle(ViewRects.Client.X, ViewRects.Client.Bottom, ViewRects.Client.Width, maxBottom - minTop), false);

            if (View.OptionsView.ShowGroupPanel)
            {
                r = ViewRects.Client;
                r.Y = minTop;
                r.Height = CalcGroupPanelHeight();
                ViewRects.GroupPanel = r;
                minTop = ViewRects.GroupPanel.Bottom;
            }
            minTop = CalcRectsColumnPanel(minTop);
            if (View.IsShowFilterPanel)
            {
                r = ViewRects.Client;
                int fPanel = GetFilterPanelHeight();
                r.Y = maxBottom - fPanel;
                r.Height = fPanel;
                FilterPanel.Bounds = r;
                maxBottom = r.Top;
            }
            if (View.OptionsView.ShowFooter)
            {
                r = ViewRects.Client;
                r.Height = GetFooterPanelHeight();
                r.Y = maxBottom - r.Height;
                ViewRects.Footer = r;
                maxBottom = r.Top;
            }
            r = ViewRects.Client;
            r.Y = minTop;
            r.Height = maxBottom - minTop;
            ViewRects.Rows = r;
        }
    }
}
